pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -305,69 +400,32 @@ func (b *Build) BuildPod(images BuildPodImages, buildContext BuildContext) (*cor
 					ifWindows(buildContext.os(), addNetworkWaitLauncherVolume())...,
 				)
 				step(
-					corev1.Container{
-						Name:      "detect",
-						Image:     b.Spec.Builder.Image,
-						Command:   []string{"/cnb/lifecycle/detector"},
-						Resources: b.Spec.Resources,
-						Args: []string{
-							"-app=/workspace",
-							"-group=/layers/group.toml",
-							"-plan=/layers/plan.toml",
-						},
-						VolumeMounts: volumeMounts([]corev1.VolumeMount{
-							layersVolume,
-							platformVolume,
-							workspaceVolume,
-						}, bindingVolumeMounts),
-						ImagePullPolicy: corev1.PullIfNotPresent,
-						Env: []corev1.EnvVar{
-							{
-								Name:  platformAPIEnvVar,
-								Value: platformAPI.Original(),
-							},
-						},
-					},
-					ifWindows(buildContext.os(), addNetworkWaitLauncherVolume(), useNetworkWaitLauncher(dnsProbeHost))...,
+					func() corev1.Container {
```

Comment:
I'd like to explore moving the instantiation of these containers within the pod spec (with the accepted possibility of some duplication). The step pattern was originally added to allow all the containers to be explicitly provided in the pod spec with some logic. 

I think we can have something like: 
```go
if platformAPILessThan07 {
    step(detect...)
    step(analyze..)
}else {
    step(analyze...)
    step(detect..)
}
						
```

Reply:
Happy to explore that in a future refactor though :)